options/logger_producers.go

```diff
@@ -0,0 +1,319 @@
+package options
+
+import (
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/send"
+	"github.com/pkg/errors"
+)
+
+var globalLoggerRegistry LoggerRegistry = &basicLoggerRegistry{
+	factories: map[string]LoggerProducerFactory{
+		LogDefault:       NewDefaultLoggerProducer,
+		LogFile:          NewFileLoggerProducer,
+		LogInherited:     NewInheritedLoggerProducer,
+		LogSumoLogic:     NewSumoLogicLoggerProducer,
+		LogInMemory:      NewInMemoryLoggerProducer,
+		LogSplunk:        NewSplunkLoggerProducer,
+		LogBuildloggerV2: NewBuildloggerV2LoggerProducer,
+	},
+}
+
+// GetGlobalLoggerRegistry returns the global logger registry.
+func GetGlobalLoggerRegistry() LoggerRegistry { return globalLoggerRegistry }
+
+///////////////////////////////////////////////////////////////////////////////
```

Comment:
Can we just split the different implementations into different files instead of use doc comments to separate implementations?